v2/compiler/argo.go

```diff
@@ -19,15 +19,38 @@ func Compile(job *pipelinespec.PipelineJob) (*wfapi.Workflow, error) {
 	}
 
 	// initialization
-	compiler := &workflowCompiler{}
-	compiler.wf = &wfapi.Workflow{}
-	compiler.templates = make(map[string]*wfapi.Template)
-	wf := compiler.wf
-	wf.APIVersion = "argoproj.io/v1alpha1"
-	wf.Kind = "Workflow"
-	wf.GenerateName = spec.GetPipelineInfo().GetName() + "-"
-	wf.Spec.ServiceAccountName = "pipeline-runner"
-	wf.Spec.Entrypoint = rootComponentName
+	wf := &wfapi.Workflow{
+		TypeMeta: k8smeta.TypeMeta{
+			APIVersion: "argoproj.io/v1alpha1",
+			Kind:       "Workflow",
+		},
+		ObjectMeta: k8smeta.ObjectMeta{
+			GenerateName: spec.GetPipelineInfo().GetName() + "-",
+			// TODO(Bobgy): uncomment the following. Temporarily commented for development purposes.
+			// Annotations: map[string]string{
+			// 	"pipelines.kubeflow.org/v2_pipeline": "true",
+			// },
+		},
+		Spec: wfapi.WorkflowSpec{
+			PodMetadata: &wfapi.Metadata{
+				Annotations: map[string]string{
+					"pipelines.kubeflow.org/v2_component": "true",
+				},
+				Labels: map[string]string{
+					"pipelines.kubeflow.org/v2_component": "true",
```

Comment:
Do we need to differentiate between the label and annotation between v2 and v2 compatible so that in the future we can figure out how many pipeline runs are using v2 and how many pipeline runs are using v2 compatible?

Reply:
+1, I think this is a good point.
Tracked in v2 project: https://github.com/kubeflow/pipelines/projects/9#card-65782240

Reply:
I don't want to change in this PR, because UI etc already recognize these annotations/labels, so that I could start verifying the UI integration.